compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -5636,6 +5662,97 @@ private STNode parseObjectMethodOrField(SyntaxKind nextTokenKind, SyntaxKind nex
         return parseObjectMethodOrField(solution.tokenKind, nextTokenKind, metadata, visibilityQualifier);
     }
 
+    /**
+     * Parse function qualifiers.
+     *
+     * @return Parsed node
+     */
+    private STNode parseFunctionQualifiers(ParserRuleContext context, List<STNode> qualifierList) {
+        STToken nextToken = peek();
+        return parseFunctionQualifiers(nextToken.kind, context, qualifierList);
+    }
+
+    private STNode parseFunctionQualifiers(SyntaxKind tokenKind, ParserRuleContext context,
+                                               List<STNode> qualifierList) {
+        while (!isEndOfFunctionQualifiers(tokenKind)) {
+            STNode qualifier;
+            switch (tokenKind) {
+                case REMOTE_KEYWORD:
+                    qualifier = parseRemoteKeyword();
+                    break;
+                case TRANSACTIONAL_KEYWORD:
+                    qualifier = parseTransactionalKeyword();
+                    break;
+                case RESOURCE_KEYWORD:
+                    qualifier = parseResourceKeyword();
+                    break;
+                default:
+                    Solution solution = recover(peek(), context, context, qualifierList);
```

Comment:
these qualifiers are always optional right? Do we need to go to the recovery?

Reply:
Yes there can be `invalid tokens` before qualifier or missing `function` keyword. Those things will recover here.

Reply:
I see. btw In case there's a missing function key we only skip the while loop. And do the qualifier validation again. 
This will be a problem, specially when there's an error in the qualifier. 
e.g  For transactional  transactional <missing function>, updateLastNodeInListWithInvalidNode will call twice for same error ryt?
I think we can simply return if the solution.kind is function keyword. WDYT?

Reply:
Validation is inside the while loop, so we do simply return if the function keyword recovered.